src/styles/_catalog.scss

```diff
@@ -1,22 +1,29 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-template-columns: repeat(4, 200px);
+  grid-template-rows: repeat(auto-fill, 300px);
```

Comment:
Four colums should start at 1024px not 1023px! (see the screnshoot). And I see the same situation for all cases.
Maybe you will come up with a better idea how to do it correct)
https://prnt.sc/Z4QRtW4pwEh_